pkg/sensors/tracing/genericlsm.go

```diff
@@ -0,0 +1,465 @@
+		argPrinters: argSigPrinters,
+		hook:        f.Hook,
+		tableId:     idtable.UninitializedEntryID,
+		policyName:  in.policyName,
+		message:     msgField,
+		tags:        tagsField,
+	}
+
+	// Parse Filters into kernel filter logic
+	lsmEntry.selectors, err = selectors.InitKernelSelectorState(f.Selectors, f.Args, nil, nil, in.selMaps)
+	if err != nil {
+		return errFn(err)
+	}
+
+	genericLsmTable.AddEntry(&lsmEntry)
+	config.FuncId = uint32(lsmEntry.tableId.ID)
+
+	lsmEntry.pinPathPrefix = sensors.PathJoin(in.sensorPath, fmt.Sprintf("glsm-%d", lsmEntry.tableId.ID))
+
+	logger.GetLogger().
+		WithField("hook", lsmEntry.hook).
+		Infof("Added lsm Hook")
+
+	return lsmEntry.tableId, nil
+}
+
+func createGenericLsmSensor(
+	spec *v1alpha1.TracingPolicySpec,
+	name string,
+	policyID policyfilter.PolicyID,
+	policyName string,
+) (*sensors.Sensor, error) {
+	var progs []*program.Program
+	var maps []*program.Map
+	var ids []idtable.EntryID
+	var selMaps *selectors.KernelSelectorMaps
+	var err error
+
+	if !kernels.MinKernelVersion("5.7") {
+		return nil, fmt.Errorf("to load LSM sensor kernel version must be greater or equal 5.7")
+	}
+
+	if !bpf.HasLSMPrograms() {
+		return nil, fmt.Errorf("enable LSM BPF by modifying the GRUB configuration /etc/default/grub with GRUB_CMDLINE_LINUX=\"lsm=bpf\"")
+	}
+
+	lsmHooks := spec.LsmHooks
+
+	in := addLsmIn{
+		sensorPath: name,
+		policyID:   policyID,
+		policyName: policyName,
+		selMaps:    selMaps,
+	}
+
+	for _, hook := range lsmHooks {
+		id, err := addLsm(&hook, &in)
+		if err != nil {
+			return nil, err
+		}
+		ids = append(ids, id)
+	}
+
+	for _, id := range ids {
+		gl, err := genericLsmTableGet(id)
+		if err != nil {
+			return nil, err
+		}
+		progs, maps = createLsmSensorFromEntry(gl, in.sensorPath, progs, maps)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &sensors.Sensor{
+		Name:  name,
+		Progs: progs,
+		Maps:  maps,
+		DestroyHook: func() error {
+			var errs error
+			for _, id := range ids {
+				_, err := genericLsmTable.RemoveEntry(id)
+				if err != nil {
+					errs = errors.Join(errs, err)
+				}
+			}
+			return errs
+		},
+	}, nil
+}
+
+func createLsmSensorFromEntry(lsmEntry *genericLsm, sensorPath string,
+	progs []*program.Program, maps []*program.Map) ([]*program.Program, []*program.Map) {
+
+	loadProgName := "bpf_generic_lsm_v57.o"
+	if kernels.EnableV61Progs() {
+		loadProgName = "bpf_generic_lsm_v61.o"
+	} else if kernels.MinKernelVersion("5.11") {
+		loadProgName = "bpf_generic_lsm_v511.o"
+	}
+
+	pinPath := lsmEntry.pinPathPrefix
+	pinProg := sensors.PathJoin(pinPath, fmt.Sprintf("%s_prog", lsmEntry.hook))
+
+	load := program.Builder(
+		path.Join(option.Config.HubbleLib, loadProgName),
+		lsmEntry.hook,
+		"lsm/generic_lsm",
+		pinProg,
+		"generic_lsm").
+		SetLoaderData(lsmEntry.tableId)
+	progs = append(progs, load)
+
+	fdinstall := program.MapBuilderPin("fdinstall_map", sensors.PathJoin(sensorPath, "fdinstall_map"), load)
+	maps = append(maps, fdinstall)
```

Comment:
It would have been nice to use the new interface, but it's not clear when it's going to be ready: https://github.com/cilium/tetragon/pull/2128.

CC: @olsajiri